openstack-installer/inventory/inventory.py

```diff
@@ -56,10 +57,18 @@ def inventory(hostname):
 
         print json.dumps(inventory, indent=4)
 
-parser = argparse.ArgumentParser(
-    description='Dynamic inventory for Openstack.')
-parser.add_argument('--list', help='list the hosts', action='store_true')
-parser.add_argument('--host', help='returnt the variables for the hosts')
-args = parser.parse_args()
 
-inventory(args.host)
+def main():
+
+    parser = argparse.ArgumentParser(
+        description='Dynamic inventory for Openstack.')
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument('--list', help='list the hosts', action='store_true')
```

Comment:
Unfortantely this gives up a nice feature: when run without any parameters, it just lists the inventory. (--list is the default)
Ansible doesn't require this, but it is convenient to see the inventory.